applications/lib_app/rpc/rpcMessageTime.cpp

```diff
@@ -0,0 +1,266 @@
+                    return false;
+                }
+
+                if (rpcMessage::sendInterruptRetry(fd, buff_ptr.get(), offset) != true) {
+                    return false;
+                }
+
+                break;
+            }
+            default:
+                break;
+        }
+
+        return true;
+    }
+
+    bool rpcMessageTime::deserialize(int fd)
+    {
+        uint16_t tmpValue;
+        if (rpcMessage::recvInterruptRetry(fd, &tmpValue, sizeof(tmpValue)) != true) {
+            return false;
+        }
+
+        this->msgAction = app::rpcMessageTimeActionType(tmpValue);
+
+        switch (this->msgAction)
+        {
+            case app::rpcMessageTimeActionType::GET_NTP_CONFIG:
+            case app::rpcMessageTimeActionType::SET_NTP_CONFIG:
+            {
+                if (rpcMessage::recvInterruptRetry(fd, &this->msgResult, sizeof(this->msgResult)) != true) {
+                    return false;
+                }
+
+                if (rpcMessage::recvInterruptRetry(fd, &this->ntpCfg, sizeof(this->ntpCfg)) != true) {
+                    return false;
+                }
+
+                break;
+            }
+            case app::rpcMessageTimeActionType::GET_SYSTEM_TIME:
+            case app::rpcMessageTimeActionType::SET_SYSTEM_TIME:
+            {
+                if (rpcMessage::recvInterruptRetry(fd, &this->msgResult, sizeof(this->msgResult)) != true) {
+                    return false;
+                }
+
+                if (rpcMessage::recvInterruptRetry(fd, &this->systemTime, sizeof(this->systemTime)) != true) {
+                    return false;
+                }
+
+                break;
+            }
+            default:
+                break;
+        }
+
+        return true;
+    }
+
+    app::rpcMessageTimeActionType rpcMessageTime::getMsgAction() const
+    {
+        return this->msgAction;
+    }
+
+    void rpcMessageTime::setMsgAction(const rpcMessageTimeActionType action)
+    {
+        this->msgAction = action;
+    }
+
+    app::rpcMessageTimeResultType rpcMessageTime::getMsgResult() const
+    {
+        return this->msgResult;
+    }
+
+    void rpcMessageTime::setMsgResult(const app::rpcMessageTimeResultType result)
+    {
+        this->msgResult = result;
+    }
+
+    struct tm const &rpcMessageTime::getSystemTime() const
+    {
+        return this->systemTime;
+    }
+
+    void rpcMessageTime::setSystemTime(const struct tm &time)
+    {
+        this->systemTime = time;
+    }
+
+    ntpConfig_t const &rpcMessageTime::getNtpCfg() const
+    {
+        return this->ntpCfg;
+    }
+
+    void rpcMessageTime::setNtpCfg(const ntpConfig_t &cfg)
+    {
+        this->ntpCfg = cfg;
+    }
+
+    app::rpcMessageTimeResultType rpcMessageTime::rpcGetSystemTime(app::rpcUnixClient &rpcClient, struct tm& time)
+    {
+        app::rpcMessageTime msg;
+
+        msg.setMsgAction(app::rpcMessageTimeActionType::GET_SYSTEM_TIME);
```

Comment:
@nguyenhien1994 , @HuuPhuc25 , pleasee follow https://github.com/hhdang6637/embedded_linux_skeleton/blob/f97a9324330dec5995c7228802f23767847a13fa/applications/lib_app/rpc/rpcMessageOpenvpn.cpp#L180

Reply:
hi a @hhdang6637 , I think it's should return an app::rpcMessageTimeResultType because we need to validate the user input and return the appropriate error code so that the user can have an intuitive view about the error. We will do that in when we implement the validate function.